Frameworks/AudioToolbox/AudioConverter.mm

```diff
@@ -16,28 +17,186 @@
 
 #import <AudioToolbox/AudioConverter.h>
 #import <StubReturn.h>
+#import <NSBundleInternal.h>
+#import <NSLogging.h>
+#import "AssertARCEnabled.h"
+
+#include <COMIncludes.h>
+#import <wrl\client.h>
+#import <wrl\wrappers\corewrappers.h>
+#include <mfapi.h>
+#include <mfidl.h>
+#include <mferror.h>
+#include <wmcodecdsp.h>
+#include <COMIncludes_End.h>
+
+using namespace Microsoft::WRL;
+using namespace Microsoft::WRL::Wrappers;
+
+struct OpaqueAudioConverter {
+    IMFTransform* transform;
+    float sizeChangeMultiplier;
+};
+
+static const wchar_t* TAG = L"AudioConverter";
 
 /**
- @Status Stub
+ @Status Caveat: AudioConverter supported only for conversions through AudioConverterConvertBuffer.
+                 Returns kAudioConverterErr_UnspecifiedError on failure and noErr (0) on success. Other return types not supported.
  @Notes
 */
 OSStatus AudioConverterDispose(AudioConverterRef inAudioConverter) {
-    UNIMPLEMENTED();
-    return StubReturn();
+    HRESULT status;
+
+    status = MFShutdown();
+    if (!SUCCEEDED(status)) {
+        NSTraceInfo(TAG, @"MFShutdown Failed");
+        return kAudioConverterErr_UnspecifiedError;
+    }
+
+    CoUninitialize();
+    delete inAudioConverter;
```

Comment:
>inAudioConverter [](start = 11, length = 16)

Calling delete on passed in memory is not a scalable memory management strategy. This creates CRT / heap dependencies between the allocation call and this call. Take for instance if someone made a derived AudioCOnverterRef etc. Given that the type is called Foo*Ref* there is a certain expectation about its memory management as well that a straight delete is likely violating. Consider instead calling release on this object and making it a real CF or ObjC object under the convers.